[AbstractList.hpp] size_t not found in 3.0.0

Hello,
I just installed your List library (version 3.0.0). When I just add #include <List.hpp> and try to compile I am getting errors.


In file included from C:\Users\User\Documents\Arduino\libraries\List\src/SingleLinkedList.hpp:29,
                 from C:\Users\User\Documents\Arduino\libraries\List\src/List.hpp:46,
                 from C:\Users\User\Documents\Arduino\libraries\Grove_3Axis_Compass_V2.0_BMM150\examples\compass\compass.ino:1:
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:36:3: error: 'size_t' does not name a type
   size_t size = 0;         /// Size of the list.
   ^~~~~~
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:36:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:1:1:
+#include <cstddef>
 /*!
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:36:3:
   size_t size = 0;         /// Size of the list.
   ^~~~~~
In file included from C:\Users\User\Documents\Arduino\libraries\List\src/SingleLinkedList.hpp:29,
                 from C:\Users\User\Documents\Arduino\libraries\List\src/List.hpp:46,
                 from C:\Users\User\Documents\Arduino\libraries\Grove_3Axis_Compass_V2.0_BMM150\examples\compass\compass.ino:1:
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:209:40: error: 'size_t' does not name a type
   void addAll(int index, T *arr, const size_t arrSize) {
                                        ^~~~~~
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:209:40: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:222:29: error: 'size_t' does not name a type
   void addAll(T *arr, const size_t arrSize) {
                             ^~~~~~
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:222:29: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
In file included from C:\Users\User\Documents\Arduino\libraries\List\src/SingleLinkedList.hpp:29,
                 from C:\Users\User\Documents\Arduino\libraries\List\src/List.hpp:46,
                 from C:\Users\User\Documents\Arduino\libraries\Grove_3Axis_Compass_V2.0_BMM150\examples\compass\compass.ino:1:
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp: In member function 'void AbstractList<T>::increaseSize()':
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:107:25: error: 'size' was not declared in this scope
   void increaseSize() { size++; }
                         ^~~~
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp: In member function 'void AbstractList<T>::decreaseSize()':
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:113:25: error: 'size' was not declared in this scope
   void decreaseSize() { size--; }
                         ^~~~
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp: In member function 'void AbstractList<T>::resetSize()':
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:118:22: error: 'size' was not declared in this scope
   void resetSize() { size = 0; }
                      ^~~~
In file included from C:\Users\User\Documents\Arduino\libraries\List\src/SingleLinkedList.hpp:29,
                 from C:\Users\User\Documents\Arduino\libraries\List\src/List.hpp:46,
                 from C:\Users\User\Documents\Arduino\libraries\Grove_3Axis_Compass_V2.0_BMM150\examples\compass\compass.ino:1:
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp: In member function 'void AbstractList<T>::addAll(int, T*, int)':
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:210:10: error: 'size_t' was not declared in this scope
     for (size_t i = 0; i < arrSize; ++i) {
          ^~~~~~
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:210:10: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:210:24: error: 'i' was not declared in this scope
     for (size_t i = 0; i < arrSize; ++i) {
                        ^
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp: In member function 'void AbstractList<T>::addAll(T*, int)':
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:223:10: error: 'size_t' was not declared in this scope
     for (size_t i = 0; i < arrSize; ++i) {
          ^~~~~~
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:223:10: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:223:24: error: 'i' was not declared in this scope
     for (size_t i = 0; i < arrSize; ++i) {
                        ^
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp: In member function 'int AbstractList<T>::getSize() const':
C:\Users\User\Documents\Arduino\libraries\List\src/AbstractList.hpp:332:32: error: 'size' was not declared in this scope
   int getSize() const { return size; }
                                ^~~~

exit status 1

Compilation error: exit status 1



with version 2.1.4 there are no errors when #include <List.hpp> .

--- a/.gitignore
+++ b/.gitignore
@@ -1,16 +1,15 @@
-# CLion
-.idea
+# Jetbrains IDEs
+**/.idea/**
 
 # CMake
-CMakeLists.txt
-cmake-build-*
+**/cmake-build-*
 
 # gh_pages files
 doxygen_sqlite3.db
 html
 
 # VSCode
-.vscode
+**/.vscode/settings.json
 
 # PlatformIO
 .pio
@@ -19,3 +18,6 @@ src/main.cpp
 include
 platformio.ini
 CMakeListsPrivate.txt
+
+# macOS
+**/.DS_Store

--- a/Doxyfile
+++ b/Doxyfile
@@ -38,7 +38,7 @@ PROJECT_NAME           = "Arduino List Library"
 # could be handy for archiving the generated documentation or if some version
 # control system is used.
 
-PROJECT_NUMBER         = 3.0.0
+PROJECT_NUMBER         = 3.0.1
 
 # Using the PROJECT_BRIEF tag one can provide an optional one line description
 # for a project that appears at the top of each page and should give viewer a

--- a/library.json
+++ b/library.json
@@ -1,7 +1,7 @@
 {
   "$schema": "https://raw.githubusercontent.com/platformio/platformio-core/develop/platformio/assets/schema/library.json",
   "name": "List",
-  "version": "3.0.0",
+  "version": "3.0.1",
   "description": "The Ultimate Collection of Lists. This library extends the Arduino ecosystem with the functionality of several easy-to-use lists for numerous purposes.",
   "keywords": [
     "arduino",

--- a/library.properties
+++ b/library.properties
@@ -1,5 +1,5 @@
 name=List
-version=3.0.0
+version=3.0.1
 author=Niklas Kaaf <nkaaf@protonmail.com>
 maintainer=Niklas Kaaf <nkaaf@protonmail.com>
 sentence=The Ultimate Collection of Lists

--- a/src/AbstractList.hpp
+++ b/src/AbstractList.hpp
@@ -5,7 +5,7 @@
  * easy-to-use list implementations. They are specially designed and optimized
  * for different purposes.
  *
- * Copyright (C) 2021-2023  Niklas Kaaf
+ * Copyright (C) 2021-2024  Niklas Kaaf
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -25,6 +25,8 @@
 
 #ifndef LIST_ABSTRACT_LIST_HPP
 #define LIST_ABSTRACT_LIST_HPP
+
+#include <stddef.h>
 
 /*!
  * @brief   Abstract class from which all lists can be derived.

